lib/vrl/stdlib/src/to_timestamp.rs

```diff
@@ -2,18 +2,29 @@ use chrono::{TimeZone as _, Utc};
 use vector_common::{conversion::Conversion, TimeZone};
 use vrl::prelude::*;
 
-fn to_timestamp(value: Value) -> Resolved {
+fn to_timestamp(value: Value, unit: Bytes) -> Resolved {
```

Comment:
Not a big deal, but I might convert `unit` to a Rust enum before this function is called to avoid the `unreachable` here.

Reply:
👍 that's probably cleaner and more closely mirrors the `to_unix_timestamp`